backend/scripts/test_citation_builder.py

```diff
@@ -0,0 +1,281 @@
+import argparse
+
+# import unittest
+import os
+import sys
+from datetime import datetime
+from typing import List
+from typing import Union
+import pytest
+
+# Modify sys.path
+current_dir = os.path.dirname(os.path.abspath(__file__))
+parent_dir = os.path.dirname(current_dir)
+sys.path.append(parent_dir)
+
+
+# pylint: disable=E402
+# flake8: noqa: E402
+from danswer.configs.constants import DocumentSource
+from danswer.llm.answering.stream_processing.citation_processing import (
+    extract_citations_from_stream,
+)
+
+from danswer.chat.models import CitationInfo
+from danswer.chat.models import DanswerAnswerPiece
+from danswer.chat.models import LlmDoc
+
+# pylint: enable=E402
+# flake8: noqa: E402
+import random
+
+
+def create_mock_tokens(text: str) -> list[str]:
```

Comment:
lists are iterable so you can just mock specific test strings by passing lists of "tokens"